[channel-picker] First iteration

DmitrySharabin opened the pull request:
Partially addresses #124.

It's a bare minimum, just for a start.

Docs: https://deploy-preview-147--color-elements.netlify.app/src/channel-picker/

LeaVerou commented on src/channel-picker/channel-picker.js:
```diff
@@ -0,0 +1,109 @@
+
+	static props = {
+		value: {
+			default: "oklch.l",
```
We should definitely have props that return the resolved ColorSpace object and the coordinate used

DmitrySharabin replied:
During the development, I was adding and removing them several times. I decided if you tell me that we need it, I'll bring them back. 🙂

LeaVerou commented on src/channel-picker/channel-picker.js:
```diff
@@ -0,0 +1,109 @@
+
+	handleEvent (event) {
+		if ([this._el.space_picker, this._el.picker].includes(event.target)) {
+			if (event.type === "spacechange") {
```
This only applies to space_picker, so I'd move it outside the if

DmitrySharabin replied:
I have nothing against it. Will do.

LeaVerou commented on src/channel-picker/channel-picker.js:
```diff
@@ -0,0 +1,109 @@
+		this._el.space_picker.removeEventListener("spacechange", this);
+		this._el.picker.removeEventListener("change", this);
+	}
+
```
```suggestion

	get selectedSpace () {
		return this._el.space_picker.selectedSpace;
	}
```

LeaVerou commented on src/channel-picker/channel-picker.js:
```diff
@@ -0,0 +1,109 @@
+		let channel = this.value?.split(".")[1];
+		if (channel && channel in coords) {
+			// Preserve the channel if it exists in the new space
+			this._el.picker.value = channel;
```
Likely post-MVP, but even if it doesn't, it would be nice to still keep it around somewhere so that switching _back_ to that space preserves your selection

DmitrySharabin replied:
Good idea! Thanks.

LeaVerou reviewed (changes requested):
Great job, only minor comments. 

One thing that should absolutely change though is the styling: It should be obvious that these are two separate selects. They should still be readable as a unit, but their styling should have **affordances** that indicate that they are separate dropdowns. Like, the triangle should still be there, for both of them. Then you also don't need the weird dashed line.

DmitrySharabin commented:
I addressed all your feedback. Thank you!

I also made the picker more resistant to and tolerant of user errors. I would be grateful for your following review.

LeaVerou reviewed (approved):
Code looks good, only remaining comment is styling: The control still tries to look like a single thing, which looks weird. I'd render it as either separate dropdowns that are side by side or a "segmented" control, where there is a line down the middle. And the border color should be way fainter in any case.

DmitrySharabin commented:
> I'd render it as either separate dropdowns that are side by side or a "segmented" control, where there is a line down the middle. And the border color should be way fainter in any case.

Thank you for your feedback. I ended up with a "segmented" control. We can improve it with the next iterations.

<img width="989" alt="image" src="https://github.com/user-attachments/assets/9d663a93-c2fb-4199-b1a9-31c0610189f2">

LeaVerou commented:
Yeah. One issue with the current styling is that it doesn't communicate the hierarchy, that Lightness is a coordinate *of* Oklch, they look like peers.
